Projects/Application/Sources/Common/Network/Http/Http.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Created by Maxim Berezhnoy on 2019-02-12.
+// Licensed under the MIT license
+//
+//
+
+import Foundation
+
+// MARK: - RequestAdapter
+protocol RequestAdapter {
+    func adapt(urlRequest: URLRequest) -> URLRequest
+}
+
+// MARK: - RequestRetrier
+protocol RequestRetrier {
+    func should(retry request: Http.Request, with response: Http.Response, retryAfter: (TimeInterval) -> Void)
+}
+
+// MARK: - Http core
+final class Http {
+    enum Method {
+        case get
+        case put
+        case post
+        case delete
+    }
+
+    typealias Headers = [String: String]
+
+    var adapter: RequestAdapter?
+    var retrier: RequestRetrier?
+
+    private let urlSession: URLSession
+
+    static let `default` = {
+       Http(configuration: .default)
+    }()
+
+    init(configuration: Configuration) {
+        let sessionConfiguration = URLSessionConfiguration.default
+        sessionConfiguration.timeoutIntervalForRequest = configuration.requestTimeout
+        sessionConfiguration.httpAdditionalHeaders =
+                Configuration.defaultHeaders.merging(configuration.additionalHeaders) { _, r in r }
```

Comment:
I would extract merging to `Configuration` and maybe make a field `headers` which represents `defaultHeaders` and can be mutated to have additional headers.

Reply:
Fixed